File: server/scripts/seed.js
#!/usr/bin/env node
'use strict';

const fs = require('node:fs');
const path = require('node:path');
const { createApp } = require('../src/app');

async function main() {
  const app = await createApp({ logger: false });
  const { db, knowledge, models, tasks, config } = app.services;
  let kb = db.one("SELECT * FROM knowledge_bases WHERE workspace_id=? AND name='Ordo 示例知识库' AND status!='deleted'", config.localWorkspaceId);
  if (!kb) kb = knowledge.createKnowledgeBase({ name: 'Ordo 示例知识库', description: '用于验证真实产品闭环的公开模拟数据' });
  const dataset = db.one("SELECT * FROM datasets WHERE knowledge_base_id=? AND status!='deleted' ORDER BY created_at LIMIT 1", kb.id);
  if (!db.one("SELECT 1 AS found FROM model_connections WHERE workspace_id=? AND provider='local-extractive'", config.localWorkspaceId)) {
    await models.create({ name: '本地严格证据模型', provider: 'local-extractive', purpose: 'generation', modelId: 'ordo-local-extractive-v1' });
  }
  const sample = path.resolve(__dirname, '..', 'fixtures', 'ordo-sample-knowledge.md');
  const content = fs.readFileSync(sample);
  const source = db.one("SELECT * FROM sources WHERE workspace_id=? AND dataset_id=? AND name='Ordo 示例产品知识' AND deleted_at IS NULL ORDER BY created_at LIMIT 1", config.localWorkspaceId, dataset.id)
    || knowledge.createSource(dataset.id, { type: 'synthetic', name: 'Ordo 示例产品知识', locationHint: 'server/fixtures/ordo-sample-knowledge.md' });
  const registered = await knowledge.registerUpload(dataset.id, source.id, path.basename(sample), content, 'text/markdown');
  if (registered.task) await tasks.wait(registered.task.id, config.localWorkspaceId, 30_000);
  const latestDataset = knowledge.getDataset(dataset.id);
  if (!latestDataset.active_release_id) {
    const releaseTask = knowledge.buildRelease(dataset.id, { activate: true });
    const release = await tasks.wait(releaseTask.id, config.localWorkspaceId, 30_000);
    if (release.status !== 'succeeded') throw new Error(`Release build failed: ${release.error_message}`);
  }
  console.log(JSON.stringify({ knowledgeBaseId: kb.id, datasetId: dataset.id, documentId: registered.document.id, duplicate: registered.duplicate, activeReleaseId: knowledge.getDataset(dataset.id).active_release_id }, null, 2));
  await app.close();
}

main().catch(error => {
  console.error('[ordo-seed] failed:', error);
  process.exit(1);
});

File: serverpy/ordo/seed.py
"""Idempotent sample content for a newly installed local workspace."""
from .app import create_app


async def seed():
    app = create_app()
    async with app.router.lifespan_context(app):
        services = app.state.services
        db, k, config = services['db'], services['knowledge'], services['config']
        ws = config['localWorkspaceId']
        existing = db.one("SELECT id FROM knowledge_bases WHERE workspace_id=? AND name='Ordo 使用指南' AND status='active'", ws)
        if existing:
            print('Sample knowledge base already exists; existing data preserved.')
            return
        kb = k.create_knowledge_base({'name': 'Ordo 使用指南', 'description': '本地示例知识库'}, ws)
        dataset_id = kb['default_dataset_id']
        source = k.create_source(dataset_id, {'name': '示例使用指南', 'type': 'synthetic'}, ws)
        upload = k.register_upload(dataset_id, source['id'], 'Ordo 使用指南.md', '# Ordo 使用指南\n\nOrdo 使用 Python 和 FastAPI 提供本地知识服务。\n\n## 导入知识\n\n上传文档后完成解析，构建并激活知识版本，然后开始问答。\n\n## 引用溯源\n\n每个回答中的引用都指向不可变知识块及原始文档修订。'.encode('utf-8'), 'text/markdown', ws)
        task = await services['tasks'].wait(upload['task']['id'], ws, 120000)
        if task['status'] != 'succeeded':
            raise RuntimeError(task.get('error_message'))
        task = k.build_release(dataset_id, {'activate': True}, ws)
        task = await services['tasks'].wait(task['id'], ws, 120000)
        if task['status'] != 'succeeded':
            raise RuntimeError(task.get('error_message'))
        print('Sample knowledge base created:', kb['id'])
